src/main/java/billgates/usecases/insert_entry/InsertEntryUseCase.java

```diff
@@ -0,0 +1,32 @@
+package billgates.usecases.insert_entry;
+
+import billgates.database.QueryEntryData;
+import billgates.entities.User;
+import billgates.interface_adapters.DatabaseGateway;
+
+import java.util.List;
+
+public class InsertEntryUseCase implements InsertEntryInputPort{
+
+    private final DatabaseGateway gateway;
+
+    public InsertEntryUseCase(DatabaseGateway gateway){
+        this.gateway = gateway;
+    }
+
+    @Override
+    public void insertEntry(InsertEntryRequestModel model) {
+
+        //Query all the entries.
+        List<QueryEntryData> result = this.gateway.getBillData(User.getInstance().getCurrentBillID()).getEntries();
+
+        //Construct a QueryEntryData for new entry, and the entry id of it is the size of all entries plus one.
```

Comment:
We cannot assume the entry id is the size + 1 because we may delete entries.

I've contacted Ray about this problem, and he may provide a new constructor without the entry id as a parameter.

Reply:
Okay!